### PR TITLE
Simulate resolution for Iqt Monte Carlo Error calculation

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR fixes a bug where the error value for the first bin in the monte carlo error calculation was zero.

The sample is randomized within its y error values, and now the resolution is also randomized within its y error values. During normalisation, the sample is divided by the resolution. Previously, this would generate a y value in the first bin with value 1, and an error of 0.

In the latest changes, the y value in the first bin is still 1, but the error is now non-zero. It makes sense that the errors in the randomized sample and resolution counts should be propagated to the result of the normalisation.

To test this change, I have done some fitting on the I(Q, t) tab of QENS Fitting. The new fits have a much better chi squared now that the first bin has a non-zero error.

*There is no associated issue.*

### To test:
1. Open Inelastic->Data Processor interface
2. Load the `irs26176_graphite002_red` as sample from the sample data
3. Load the `irs26173_graphite002_res` as resolution from the sample data
4. Tick `Enforce Normalisation`.
5. Click Run. Wait. Plot the first spectrum of the result `_iqt` workspace with error bars:
image
6. Notice the error for the first bin is non-zero
7. Untick `Enforce Normalisation`.
8. Click Run. Wait. Plot the first spectrum of the result `_iqt` workspace with error bars:
image
9. Tick `Enforce Normalisation`.
10. Click Run. 
11. Open Inelastic-> QENS Fitting
12. Load the `_iqt` workspace from the ADS
13. Select `Fit Type` to be Stretched Exp
14. Select `Background` to be `Flat Background`
15. Click Run and wait.
16. The Chi squared above the Function Browser should be roughly 1.28 . Previously this was higher

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Comment by robertapplin** [comment]
@mpatrou would you be happy to check I haven't broken the `EnforceNormalisation` option you added recently in https://github.com/mantidproject/mantid/pull/37005 ?

**Comment by mpatrou** [comment]
> @mpatrou would you be happy to check I haven't broken the `EnforceNormalisation` option you added recently in #37005 ?

Absolutely! I will review it.

File: Framework/Algorithms/src/CalculateIqt.cpp [hunk 1]
```diff
@@ -269,28 +272,31 @@ MatrixWorkspace_sptr CalculateIqt::normalizedFourierTransform(MatrixWorkspace_sp
...
-MatrixWorkspace_sptr CalculateIqt::doSimulation(MatrixWorkspace_sptr sample, const MatrixWorkspace_sptr &resolution,
+MatrixWorkspace_sptr CalculateIqt::doSimulation(MatrixWorkspace_sptr sample, MatrixWorkspace_sptr resolution,
                                                 const std::string &rebinParams, MersenneTwister &mTwister,
                                                 const bool enforceNormalization) {
-  auto simulatedWorkspace = randomizeWorkspaceWithinError(std::move(sample), mTwister);
-  return calculateIqt(simulatedWorkspace, resolution, rebinParams, enforceNormalization);
+  auto const simulatedSample = randomizeWorkspaceWithinError(std::move(sample), mTwister);
+  auto const simulatedResolution = randomizeWorkspaceWithinError(std::move(resolution), mTwister);
+  auto const iqt = calculateIqt(simulatedSample, simulatedResolution, rebinParams, false);
+  return enforceNormalization ? divide(iqt, m_sampleIntegral) : iqt;
```
Comment (mpatrou) [inline Framework/Algorithms/src/CalculateIqt.cpp:299]: return divide(iqt, m_sampleIntegral);

File: Framework/Algorithms/src/CalculateIqt.cpp [hunk 2]
```diff
@@ -138,7 +138,10 @@ void CalculateIqt::exec() {
   bool enforceNormalization = getProperty("EnforceNormalization");
   const int nIterations = getProperty("NumberOfIterations");
   const int seed = getProperty("SeedValue");
-  resolution = normalizedFourierTransform(resolution, rebinParams, enforceNormalization);
+
+  if (enforceNormalization) {
```
Comment (mpatrou) [inline Framework/Algorithms/src/CalculateIqt.cpp:142]: this can be omitted

Comment (mpatrou) [inline Framework/Algorithms/src/CalculateIqt.cpp:142]: The whole if block can be omitted

File: Framework/Algorithms/src/CalculateIqt.cpp [hunk 3]
```diff
@@ -269,28 +272,31 @@ MatrixWorkspace_sptr CalculateIqt::normalizedFourierTransform(MatrixWorkspace_sp
...
   return workspace;
 }
 
 MatrixWorkspace_sptr CalculateIqt::calculateIqt(MatrixWorkspace_sptr workspace,
-                                                const MatrixWorkspace_sptr &resolutionWorkspace,
+                                                MatrixWorkspace_sptr resolutionWorkspace,
                                                 const std::string &rebinParams, const bool enforceNormalization) {
   workspace = normalizedFourierTransform(workspace, rebinParams, enforceNormalization);
+  // Always normalise the resolution
+  resolutionWorkspace = normalizedFourierTransform(resolutionWorkspace, rebinParams, true);
```
Comment (mpatrou) [inline Framework/Algorithms/src/CalculateIqt.cpp:289]: resolutionWorkspace = normalizedFourierTransform(resolutionWorkspace, rebinParams, enforceNormalization);

**Comment by robertapplin** [comment]
Closed in favour of https://github.com/mantidproject/mantid/pull/37653

**Pull request closed without merging**